--- src/DDFacet/Other/AsyncProcessPool.py	2020-09-28 12:34:05.505711193 +0200
+++ src/DDFacet/Other/AsyncProcessPool_new.py	2020-09-28 14:07:01.553615836 +0200
@@ -274,6 +274,9 @@
         else:
             print(ModColor.Str("Fixing parent process to vthread %d" % self.parent_affinity, col="green"), file=log)
         psutil.Process().cpu_affinity(range(self.ncpu) if not self.parent_affinity else [self.parent_affinity])
+        avail_cores = psutil.Process().cpu_affinity()
+        psutil.Process().cpu_affinity(avail_cores[:self.ncpu] if not self.parent_affinity else [self.parent_affinity])
+
 
         # if NCPU is 0, set to number of CPUs on system
         if not self.ncpu:
@@ -291,7 +294,8 @@
         # create a queue for compute-bound tasks
         # generate list of CPU cores for workers to run on
         if isinstance(self.affinity, int) and (not self.affinity or self.affinity == 1):
-            cores = range(self.ncpu)
+            #cores = range(self.ncpu)
+            cores = avail_cores[:self.ncpu]
         elif isinstance(self.affinity, int) and self.affinity == 2:
             cores = range(0, self.ncpu * 2, 2)
         elif isinstance(self.affinity, int) and self.affinity == -2:
@@ -299,7 +303,8 @@
         elif isinstance(self.affinity, list):
             cores = self.affinity[:self.ncpu]
         elif not self.affinity:
-            cores = range(self.ncpu)
+            #cores = range(self.ncpu)
+            cores =  avail_cores[:self.ncpu]
         else:
             raise ValueError("unknown affinity setting")
         if not self.affinity:
